pkgs/moduleit/module-definition.nix

```diff
@@ -110,71 +110,79 @@ let
         };
     in
     {
-      options = {
-        name = mkOption {
-          type = types.str;
-          description = lib.mdDoc ''
-            The name of the runner.
-          '';
-        };
+      name = mkOption {
+        type = types.str;
+        description = lib.mdDoc ''
+          The name of the runner.
+        '';
+      };
 
-        language = mkOption {
-          type = types.str;
-          description = lib.mdDoc ''
-            The language this runner supports.
-          '';
-        };
+      language = mkOption {
+        type = types.str;
+        description = lib.mdDoc ''
+          The language this runner supports.
+        '';
+      };
 
-        start = mkOption {
-          type = commandType;
-          description = lib.mdDoc ''
-            The command to run a file. Use $file to substitute in the file path.
-          '';
-        };
+      start = mkOption {
+        type = commandType;
+        description = lib.mdDoc ''
+          The command to run a file. Use $file to substitute in the file path.
+        '';
+      };
 
-        fileParam = mkOption {
-          type = types.bool;
-          default = false;
-          description = lib.mdDoc ''
-            Whether this runner accepts a $file paramater.
-          '';
-        };
+      fileParam = mkOption {
+        type = types.bool;
+        default = false;
+        description = lib.mdDoc ''
+          Whether this runner accepts a $file paramater.
+        '';
+      };
 
-        interpreter = mkOption {
-          type = types.bool;
-          default = false;
-          description = lib.mdDoc ''
-            Whether this runner starts an interpreter. Default: false.
-          '';
-        };
+      optionalFileParam = mkOption {
```

Comment:
Nit: Is there a way to make these mutually exclusive?

e.g. `fileParam: { optional: true }`/`fileParam { optional: false }` or something?

Reply:
I added an assertion that makes sure you can set both of them https://github.com/replit/nixmodules/blob/b81148d6b7dfb904f9d894ebe9805e209efb430c/pkgs/moduleit/module-definition.nix#L606

Reply:
I see, thanks